libsplinter/src/admin/store/proposed_circuit.rs

```diff
@@ -491,3 +493,39 @@ impl ProposedCircuitBuilder {
         Ok(create_circuit_message)
     }
 }
+
+impl TryFrom<&messages::CreateCircuit> for ProposedCircuit {
+    type Error = InvalidStateError;
+
+    fn try_from(create_circuit: &messages::CreateCircuit) -> Result<ProposedCircuit, Self::Error> {
+        let services = create_circuit
+            .roster
+            .iter()
+            .map(ProposedService::try_from)
+            .collect::<Result<Vec<ProposedService>, InvalidStateError>>()?;
+        let nodes = create_circuit
+            .members
+            .iter()
+            .map(ProposedNode::from)
+            .collect::<Vec<ProposedNode>>();
+        let mut circuit_builder = ProposedCircuitBuilder::new()
+            .with_circuit_id(&create_circuit.circuit_id)
+            .with_roster(&services)
+            .with_members(&nodes)
+            .with_authorization_type(&AuthorizationType::from(&create_circuit.authorization_type))
+            .with_persistence(&PersistenceType::from(&create_circuit.persistence))
+            .with_durability(&DurabilityType::from(&create_circuit.durability))
+            .with_routes(&RouteType::from(&create_circuit.routes))
+            .with_circuit_management_type(&create_circuit.circuit_management_type)
+            .with_application_metadata(&create_circuit.application_metadata);
```

Comment:
I think this should check to see if its not  empty before setting.

Reply:
Updated to check if the metadata is empty before adding it